console/src/view/async_ops.rs

```diff
@@ -0,0 +1,244 @@
+        "Total",
+        "Busy",
+        "Idle",
+        "Polls",
+        "Attributes",
+    ];
+
+    fn render<B: tui::backend::Backend>(
+        table_list_state: &mut TableListState<Self>,
+        styles: &view::Styles,
+        frame: &mut tui::terminal::Frame<B>,
+        area: layout::Rect,
+        state: &mut State,
+        ctx: Self::Context,
+    ) {
+        let now = if let Some(now) = state.last_updated_at() {
+            now
+        } else {
+            // If we have never gotten an update yet, skip...
+            return;
+        };
+
+        let AsyncOpsTableCtx {
+            initial_render,
+            resource_id,
+        } = ctx;
+
+        if initial_render {
+            table_list_state
+                .sorted_items
+                .extend(state.async_ops_state().async_ops().filter(|op| {
+                    op.upgrade()
+                        .map(|op| resource_id == op.borrow().resource_id())
+                        .unwrap_or(false)
+                }))
+        } else {
+            table_list_state.sorted_items.extend(
+                state
+                    .async_ops_state_mut()
+                    .take_new_async_ops()
+                    .filter(|op| {
+                        op.upgrade()
+                            .map(|op| resource_id == op.borrow().resource_id())
+                            .unwrap_or(false)
+                    }),
+            )
+        };
+        table_list_state
+            .sort_by
+            .sort(now, &mut table_list_state.sorted_items);
+
+        let mut id_width = view::Width::new(Self::HEADER[0].len() as u16);
+        let mut parent_width = view::Width::new(Self::HEADER[1].len() as u16);
+        let mut task_width = view::Width::new(Self::HEADER[2].len() as u16);
+        let mut source_width = view::Width::new(Self::HEADER[3].len() as u16);
+        let mut polls_width = view::Width::new(Self::HEADER[7].len() as u16);
+
+        let dur_cell = |dur: std::time::Duration| -> Cell<'static> {
+            Cell::from(styles.time_units(format!(
+                "{:>width$.prec$?}",
+                dur,
+                width = DUR_LEN,
+                prec = DUR_PRECISION,
+            )))
+        };
+
+        let rows = {
+            let id_width = &mut id_width;
+            let parent_width = &mut parent_width;
+            let task_width = &mut task_width;
+            let source_width = &mut source_width;
+            let polls_width = &mut polls_width;
+
+            table_list_state
+                .sorted_items
+                .iter()
+                .filter_map(move |async_op| {
+                    let async_op = async_op.upgrade()?;
+                    let async_op = async_op.borrow();
+
+                    let parent_id_str = match async_op.parent_id() {
+                        Some(id) => {
+                            format!("{}", id)
+                        }
+                        None => "n/a".to_string(),
```

Comment:
it weould be nice to avoid re-formatting and allocating these strings every time...